docs/concepts/aggregation.md

```diff
@@ -0,0 +1,165 @@
+a sample addon API server which serves an API group called `wardle`, with
+an single namespaced resource `flunders`, abbreviated `fl`.
+
+Understanding how the aggregator works with addon API servers requires
+understanding how [authentication and authorization](./auth.md) works.  If
+you have not yet read that section, please do.
+
+## Enabling the Aggregator
+
+The API aggregator is integrated into the main Kubernetes API server in
+Kubernetes 1.7+, but must be run as a separate pod in Kubernetes 1.6.  In
+Kubernetes 1.7+, it is only served on the secure port of the API server.
+
+In Kubernetes 1.7, in order for the aggregator to work properly, it needs
+to have the appropriate certificates.  First, ensure that you have
+RequestHeader CA certificates, and client certificates signed by that CA,
+as discussed in the [authentication and authorization](./auth.md) section.
+
+These certificates must be passed to the main Kubernetes API server using
+the flags `--proxy-client-cert-file` and `--proxy-client-key-file`.  This
+allows the aggregator to identify itself when making requests, so that
+addon API servers can use its delegated RequestHeader authentication.
+
+Enabling the aggregator in 1.6 is outside the scope of this document.  The
+aggregator directory itself provides examples on how to do so in the
+Kubernetes 1.6 release.
+
+## kubectl and Discovery
+
+When `kubectl get fl` is called, `kubectl` does not initially know what an
+`fl` is, or how to get one.  In order to determine this information, it
+uses a mechanism called *discovery*.
+
+All Kubernetes API servers serve discovery information.  To get this
+information, `kubectl` first queries `https://$SERVER/apis`, which lists
+all available API groups and versions, as well as the *preferred version*.
+Then, for each of these API groups and versions, it queries
+`https://$SERVER/apis/$GROUP/$VERSION`.  This returns a list of resources,
+along with whether the resource is *namespaced*, which operations (e.g.
+`get` or `list`) it supports, as well as any short names (like `fl` for
+`fluders`).
+
+Once `kubectl` has retrieved the entire set of available resources for
+a particular cluster, it can then determine how to operate on the exposed
+resources.  For example, in our case above, `kubectl` determines that `fl`
+is a shortname for `flunders`, and that `fluders` are a namespaced
+resource that's part of the API group `wardle`, which has a preferred
+version of `v1alpha`.  Thus, `kubectl` will attempt a query of
+`https://$SERVER/apis/wardle/v1alpha1/namespaces/$NS/flunders/`.
+
+Since querying all of these discovery endpoints for every kubectl request
+would be expensive, `kubectl` caches this information in the same folder
+as where your kubeconfig goes, based on the name of the cluster to which
+you are connecting. Clearing this cache can resolve issues with not being
+able to use `kubectl` to fetch resources from newly registered APIs.
```

Comment:
interesting.  I've had this explicitly not work properly before with aggregation.